### PR TITLE
[MAINTENANCE] Retire `docs-integration` tests that are already running in GitHub Actions

### PR DESCRIPTION
**Pull request by Shinnnyshinshin**
# What has changed? 
* `docs-snippet` tests were running on both `ci.yml` and `ci/azure-pipelines-docs-integration.yml`. Remove tests in Azure pipeline that have already been migrated over to `ci.yml`
* Add `spark` tests to `docs-snippet` tests
* Add `docs-changed `condition to `docs-snippet` stage, which was the case in Azure and had not been migrated over. 

- [x] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [x] Code is linted - run `invoke lint` (uses `black` + `ruff`)
- [x] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -107,7 +107,7 @@ jobs:
 
   docs-snippets:
     name: docs-snippets
-    if: github.event.pull_request.draft == false
+    if: (github.event.pull_request.draft == false && steps.changed-files.outputs.docs == 'true')
```
Comment (Shinnnyshinshin) [inline .github/workflows/ci.yml]: This condition had not been migrated over yet

Comment (Shinnnyshinshin) [reply .github/workflows/ci.yml]: It's actually what makes me a little nervous though. 

So in our current Azure pipeline, we run the same scope check that has now been added to the entire `docs-snippet` stage. https://github.com/great-expectations/great_expectations/blob/748fc89a21d8bfa7c29d40684be993037c217f49/ci/azure-pipelines-docs-integration.yml#L49 

This means the stage is only run for PRs where docs-related files (like `docs/**`) are changed. That is all fine and good. Here is my question:  Did this also mean that our release pipeline will not run these tests unless something has changed? Is that ok?

Comment (Kilo59) [reply .github/workflows/ci.yml]: Can you do an or here?

Run if docs-changed or event is not pull_request

Comment (Shinnnyshinshin) [reply .github/workflows/ci.yml]: Great idea!

**Review by Kilo59 — APPROVED** [review]

**Pull request merged**